app/models/foreman_tasks/task.rb

```diff
@@ -77,6 +77,14 @@ def paused?
       self.state == 'paused'
     end
 
+    def with_parents
+      [self].tap do |ret|
+        if parent_task
+          ret.concat(parent_task.with_parents)
+        end
+      end
```

Comment:
If you would like to make things more complicated `[self, *(parent_task.with_parents if parent_task)]` ;)